fdbserver/CoordinatedState.actor.cpp

```diff
@@ -205,24 +195,37 @@ struct CoordinatedStateImpl {
 				maxGen = std::max(maxGen, wrep_reply[i].get());
 		return maxGen;
 	}
-};
 
-CoordinatedState::CoordinatedState(ServerCoordinators const& coord) : impl(new CoordinatedStateImpl(coord)) {}
-CoordinatedState::~CoordinatedState() {
-	delete impl;
-}
-Future<Value> CoordinatedState::read() {
-	return CoordinatedStateImpl::read(impl);
-}
-Future<Void> CoordinatedState::onConflict() {
-	return CoordinatedStateImpl::onConflict(impl);
-}
-Future<Void> CoordinatedState::setExclusive(Value v) {
-	return CoordinatedStateImpl::setExclusive(impl, v);
-}
-uint64_t CoordinatedState::getConflict() {
-	return impl->getConflict();
-}
+public:
+	// Callers must ensure that any outstanding operations have been cancelled before destructing *this!
+	CoordinatedState(ServerCoordinators const& c)
+	  : coordinators(c), stage(0), conflictGen(0), doomed(false), ac(false), initial(false) {}
+
+	Future<Value> read() { return read(this); }
+	// May only be called once.
```

Comment:
I think we should start moving the comment-below declaration to comment-above declaration style when possible.

Reply:
I agree, I'll make this change in this file.